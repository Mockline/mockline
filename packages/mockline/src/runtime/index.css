@import "tailwindcss";
@import "assets/css/animation.css";

@variant dark (&:where(.dark, .dark *));

@plugin "@tailwindcss/typography";

@theme {
  --color-neutral: var(--ui-bg);
  --color-neutral-elevated: var(--ui-bg-elevated);
  --color-border: var(--ui-border);
}

@layer base {
  :root {
    color-scheme: light;

    --ui-text-dimmed: var(--ui-color-neutral-400);
    --ui-text-muted: var(--ui-color-neutral-500);
    --ui-text-toned: var(--ui-color-neutral-600);
    --ui-text: var(--ui-color-neutral-700);
    --ui-text-highlighted: var(--ui-color-neutral-900);

    --ui-bg: var(--color-white);
    --ui-bg-elevated: var(--ui-color-neutral-100);
    --ui-bg-accented: var(--ui-color-neutral-200);
    --ui-bg-inverted: var(--ui-color-neutral-900);

    --ui-border: var(--ui-color-neutral-200);
    --ui-border-accented: var(--ui-color-neutral-300);
    --ui-border-inverted: var(--ui-color-neutral-900);

    --ui-radius: var(--radius);

    /*--ui-slate: var(--color-slate-500);
    --ui-gray: var(--color-gray-500);
    --ui-zinc: var(--color-zinc-500);
    --ui-stone: var(--color-stone-500);
    --ui-neutral: var(--color-neutral-500);
    --ui-red: var(--color-red-500);
    --ui-orange: var(--color-orange-500);
    --ui-amber: var(--color-amber-500);
    --ui-yellow: var(--color-yellow-500);
    --ui-lime: var(--color-lime-500);
    --ui-green: var(--color-green-500);
    --ui-emerald: var(--color-emerald-500);
    --ui-teal: var(--color-teal-500);
    --ui-cyan: var(--color-cyan-500);
    --ui-sky: var(--color-sky-500);
    --ui-blue: var(--color-blue-500);
    --ui-indigo: var(--color-indigo-500);
    --ui-violet: var(--color-violet-500);
    --ui-purple: var(--color-purple-500);
    --ui-fuchsia: var(--color-fuchsia-500);
    --ui-pink: var(--color-pink-500);
    --ui-rose: var(--color-rose-500);*/
  }

  .dark {
    color-scheme: dark;

    --ui-text-dimmed: var(--ui-color-neutral-500);
    --ui-text-muted: var(--ui-color-neutral-400);
    --ui-text-toned: var(--ui-color-neutral-300);
    --ui-text: var(--ui-color-neutral-200);
    --ui-text-highlighted: var(--color-white);

    --ui-bg: var(--ui-color-neutral-900);
    --ui-bg-elevated: var(--ui-color-neutral-800);
    --ui-bg-accented: var(--ui-color-neutral-700);
    --ui-bg-inverted: var(--color-white);

    --ui-border: var(--ui-color-neutral-800);
    --ui-border-accented: var(--ui-color-neutral-700);
    --ui-border-inverted: var(--color-white);
  }
}

