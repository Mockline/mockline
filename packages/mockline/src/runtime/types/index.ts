import type { Color } from '../utils/colors'

export * from '../components/elements/Button.vue'

export type ModuleOptions = {
  /**
   * Prefix for all components
   */
  prefix?: string,
  /**
   * Enable Nuxt Content
   * @default false
   */
  content?: boolean
  /**
   * The colors to use
   */
  colors?: Color[]
  /**
   * Enable Color Transitions
   * @default false
   */
  transitions?: boolean
}
