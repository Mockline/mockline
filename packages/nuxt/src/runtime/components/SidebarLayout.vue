<script setup lang="ts">
import { SplitterGroup, SplitterPanel } from 'reka-ui'
import { ref } from 'vue'
import { useCookie } from '#imports'

type SidebarLayoutProps = {
  minSize?: number
  maxSize?: number
  collapsedSize?: number
  defaultSize?: number
}

const props = withDefaults(defineProps<SidebarLayoutProps>(), {
  minSize: 15,
  maxSize: 15,
  collapsedSize: 0,
  defaultSize: 20,
})

const layout = useCookie<number[]>('splitter:layout', {
  default: () => [props.defaultSize, 80],
})

const sidebar = ref<InstanceType<typeof SplitterPanel>>()


const toggleSidebar = (): void => {
  const isCollapsed = sidebar.value?.isCollapsed
  if (isCollapsed) sidebar.value?.expand()
  else
    sidebar.value?.collapse()
}

defineExpose({
  toggleSidebar
})

type SidebarLayoutSlots = {
  default(props: { toggleSidebar: () => void }): any
  sidebar: any
}

const slots = defineSlots<SidebarLayoutSlots>()
</script>

<template>
  <div class="h-screen">
    <SplitterGroup
      id="sidebar-layout"
      direction="horizontal"
      auto-save-id="mockline-sidebar"
      @layout="layout = $event"
    >
      <SplitterPanel
        id="sidebar-layout-sidebar"
        ref="sidebar"
        :min-size="props.minSize"
        :max-size="props.maxSize"
        :collapsed-size="props.collapsedSize"
        :default-size="layout[0]"
        collapsible
        class="max-sm:hidden"
      >
        <slot name="sidebar" />
      </SplitterPanel>
      <SplitterPanel
        id="sidebar-layout-content"
        :min-size="50"
        :default-size="layout[1]"
        style="overflow-y: auto;"
      >
        <div id="app-main-content" class="h-[calc(100vh-var(--spacing)*2)] overflow-y-auto relative m-1 rounded-lg dark:bg-neutral-900 bg-white border dark:border-neutral-800 border-neutral-200 shadow-lg dark:shadow-xl">
          <slot name="default" :toggle-sidebar />
        </div>
      </SplitterPanel>
    </SplitterGroup>
  </div>
</template>
