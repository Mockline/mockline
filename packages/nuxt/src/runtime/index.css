@import "@mockline/themes";
@import "./mockline.css";
