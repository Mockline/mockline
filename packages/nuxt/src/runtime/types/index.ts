// Elements
export * from '../components/Button.vue'
export * from '../components/NavMenu.vue'

// Layout
export * from '../components/App.vue'
export * from '../components/Aside.vue'
export * from '../components/Card.vue'
export * from '../components/Header.vue'
export * from '../components/Main.vue'
export * from '../components/Page.vue'
export * from '../components/PageBody.vue'
export * from '../components/PageHeader.vue'

// Settings
export * from '../components/ThemeToggle.vue'

// Overlays
export * from '../components/Toasts.vue'

// Content
export * from '../components/content/ContentNavigationTree.vue'
export * from '../components/content/ContentSurround.vue'
export * from '../components/content/ContentToc.vue'
export * from '../components/content/ContentTocLinks.vue'

// Global
export * from '../components/prose/Callout.vue'
export * from '../components/prose/CodeGroup.vue'
export * from '../components/prose/ProseCode.vue'
export * from '../components/prose/CodeButton.vue'
