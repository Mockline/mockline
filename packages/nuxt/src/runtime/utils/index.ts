export function transformSlot(slot: any, index: number): any {
  if (typeof slot.type === 'symbol') {
    return slot.children?.map(transformSlot)
  }
  return {
    label: slot.props?.filename || slot.props?.label || `${index}`,
    icon: slot.props?.icon,
    component: slot
  }
}

export * from './useComponent'
