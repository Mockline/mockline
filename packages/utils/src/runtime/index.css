@import "tailwindcss";

@variant dark (&:where(.dark, .dark *));

@plugin "@tailwindcss/typography";
