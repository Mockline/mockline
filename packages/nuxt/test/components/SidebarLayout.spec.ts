import { describe, it, expect } from 'vitest'
import { renderSuspended } from '@nuxt/test-utils/runtime'
import SidebarLayout from '../../src/runtime/components/SidebarLayout.vue'

describe('SidebarLayout', () => {
  it('can mount component', async () => {
    const html = await renderSuspended(SidebarLayout)
    expect(html.html()).toMatchSnapshot()
  })
})
