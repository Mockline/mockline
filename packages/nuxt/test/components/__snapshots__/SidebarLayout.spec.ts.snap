// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SidebarLayout > renders base case correctly 1`] = `
"<div class="h-screen">
  <div style="display: flex; flex-direction: row; height: 100%; overflow: hidden; width: 100%;" data-panel-group="" data-orientation="horizontal" data-panel-group-id="sidebar-layout">
    <div id="sidebar-layout-sidebar" style="flex-basis: 0px; flex-grow: 20; flex-shrink: 1; overflow: hidden;" data-panel="" data-panel-collapsible="true" data-panel-group-id="sidebar-layout" data-panel-id="sidebar-layout-sidebar" data-panel-size="20.0" data-state="expanded" class="max-sm:hidden"></div>
    <div id="sidebar-layout-content" style="flex-basis: 0px; flex-grow: 80; flex-shrink: 1; overflow: hidden; overflow-y: auto;" data-panel="" data-panel-group-id="sidebar-layout" data-panel-id="sidebar-layout-content" data-panel-size="80.0">
      <div class="overflow-y-auto relative rounded-[calc(var(--radius-base)*2)] bg-content-area border border-content-area-border h-[calc(100vh-var(--spacing)*2)] m-1">default</div>
    </div>
  </div>
</div>"
`;

exports[`SidebarLayout > renders sizes correctly 1`] = `
"<div class="h-screen">
  <div style="display: flex; flex-direction: row; height: 100%; overflow: hidden; width: 100%;" data-panel-group="" data-orientation="horizontal" data-panel-group-id="sidebar-layout">
    <div id="sidebar-layout-sidebar" style="flex-basis: 0px; flex-grow: 10; flex-shrink: 1; overflow: hidden;" data-panel="" data-panel-collapsible="true" data-panel-group-id="sidebar-layout" data-panel-id="sidebar-layout-sidebar" data-panel-size="10.0" data-state="expanded" class="max-sm:hidden"></div>
    <div id="sidebar-layout-content" style="flex-basis: 0px; flex-grow: 90; flex-shrink: 1; overflow: hidden; overflow-y: auto;" data-panel="" data-panel-group-id="sidebar-layout" data-panel-id="sidebar-layout-content" data-panel-size="90.0">
      <div class="overflow-y-auto relative rounded-[calc(var(--radius-base)*2)] bg-content-area border border-content-area-border h-[calc(100vh-var(--spacing)*2)] m-1">default</div>
    </div>
  </div>
</div>"
`;
